src/MembershipContextFactory.php

```diff
@@ -53,14 +60,25 @@ private function newPimple(): Container {
 			return $entityManager;
 		};
 
-		$pimple['token_generator'] = function() {
+		$container['fundraising.membership.application.token_generator'] = function() {
 			return new RandomMembershipTokenGenerator(
 				$this->config['token-length'],
 				new \DateInterval( $this->config['token-validity-timestamp'] )
 			);
 		};
 
-		return $pimple;
+		$container['fundraising.membership.application.authorizer.class'] = DoctrineApplicationAuthorizer::class;
+		// @todo Consider if the tokens should not better be method parameters (see ApplicationAuthorizer interface)
+		$container['fundraising.membership.application.authorizer.update_token'] = null;
+		$container['fundraising.membership.application.authorizer.access_token'] = null;
+
+		$container['fundraising.membership.application.authorizer'] = $container->factory( function ( Container $container ): ApplicationAuthorizer {
```

Comment:
If you're using `factory`, you will get a new instance each time you request the service. I guess there is no other way if we want to make the tests independent from each other, right?

Reply:
> If you're using factory, you will get a new instance each time you request the service

It is primarily relevant in the tests, as there is no real world scenario I know of in which two authorizations are performed within the context of the same request, but not strictly limited to it. In my eyes the core question ([which I raised here](https://github.com/wmde/fundraising-memberships/pull/4/files#diff-812ec1fa31c18c9510df06339fd6cc88R71)) is the [interface of the service](https://github.com/wmde/fundraising-memberships/blob/master/src/Authorization/ApplicationAuthorizer.php) expecting something that can by dynamic (the access and update tokens) set by the constructor. I postponed that question, implementing the pimple-equivalent of what [FunFunFactory did before](https://github.com/wmde/FundraisingFrontend/pull/1196/commits/53110090184c703d12929176e468ceb927e535b3#diff-b27bd68e2f2f9595828e438d3c9d2e24R1168) - but do not like it either.

Reply:
Suggestion for convention https://symfony.com/doc/current/contributing/code/standards.html#naming-conventions ff.

Reply:
The tokens where put in the constructor to keep the interface decoupled from the authentication method (tokens). If I am not mistaken one of the BCs has such a service that has these tokens in its constructor, the other has it in the interface.

Auth is normally a supporting/generic subdomain.

In tests you construct the whole object graph, including the top level factory, fresh for every test. So no need to avoid caching in the top level factory for the sake of the tests.